AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3148,6 +3148,194 @@ function Debug-AzStorageAccountADObject
+                if (!($kerberosTicketEncryptionClient.Contains("AES-256")) -and $line.Contains("AES-256"))
+                {
+                    $kerberosTicketEncryptionClient += "AES-256"
+                }
+                if (!($kerberosTicketEncryptionClient.Contains("RC4-HMAC")) -and $line.Contains("RC4-HMAC"))
+                {
+                    $kerberosTicketEncryptionClient += "RC4-HMAC"
+                }
+            }
+
+            if ($kerberosTicketEncryptionClient.Count -eq 0)
+            {
+                Write-Error -Message "No Kerberos Ticket Encryption is supported on the client side" -ErrorAction Stop
+            }
+        }
+
+        if ($kerberosTicketEncryptionClient.Value)
+        {
+            $kerberosTicketEncryptionClient = $kerberosTicketEncryptionClient.Value.ToString().replace(' ', '') -split ','
+        }
+
+
+        $kerberosTicketEncryptionServer = $protocolSettings.KerberosTicketEncryption
+        if($null -eq $kerberosTicketEncryptionServer)
+        {
+            $kerberosTicketEncryptionServer = "RC4-HMAC", "AES-256" # null(default): all values are accepted on the server
+        }
+        $kerberosTicketEncryptionServer = [String]::Join(", ", $kerberosTicketEncryptionServer)
+        $kerberosTicketEncryptionServerNoDash = $kerberosTicketEncryptionServer.replace('-','')
+
+        Write-Verbose "Kerberos Ticket Encryption supported on the client side: $kerberosTicketEncryptionClient"
+        Write-Verbose "Kerberos Ticket Encryption supported on the server side: $kerberosTicketEncryptionServerNoDash"
+        
+        $found = $false
+        foreach($type in $kerberosTicketEncryptionClient)
+        {
+            if ($kerberosTicketEncryptionServerNoDash.Contains($type)) 
+            {
+                $found = $true
+                break
+            }
+        }
+
+        if (!$found) 
+        {
+            Write-Error -Message "The server side and the client side do not have a Kerberos Ticket Encryption type in common." -ErrorAction Stop
+        }
+
+    }
+}
+
+function Debug-ChannelEncryption
+{
+    [CmdletBinding()]
+
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Resource group name")]
+        [string]$ResourceGroupName
+    )
+
+    process
+    {
+
+        $storageAccount = Validate-StorageAccount -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
+
+        $protocolSettings = (Get-AzStorageFileServiceProperty -StorageAccount $storageAccount -ErrorAction Stop).ProtocolSettings.Smb
+
+        $cmdletNeeded = "Get-SmbServerConfiguration"
+        if(!(Get-Command $cmdletNeeded -ErrorAction SilentlyContinue))
+        {
+            Write-Error -Message "Your system does not have or support the command needed for the check '$cmdletNeeded'." -ErrorAction Stop

```

Comment:
Make this a warning and skip CheckChannelEncryption. This is not an error for customer. It is just missing a dependency.